### PR TITLE
br: Fix new collaction enable check

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33422

Problem Summary:

### What is changed and how it works?
- backup up-stream `NewCollactionEnable` into backupmeta
- check `NewCollactionEnable` in backupmeta is the same as the down-stream

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Fix the issue that NewCollationEnable config not checked during restoration.
```

### CODE REVIEW

File: br/pkg/task/restore.go [hunk 1]
```diff
@@ -265,6 +266,34 @@ func CheckRestoreDBAndTable(client *restore.Client, cfg *RestoreConfig) error {
 	return nil
 }
 
+func CheckNewCollationEnable(backupNewCollationEnable string, g glue.Glue, storage kv.Storage, CheckRequirements bool) error {
+	se, err := g.CreateSession(storage)
+	if err != nil {
+		return errors.Trace(err)
+	}
```
Comment (3pointer) [inline br/pkg/task/restore.go:273]: move to L283

File: br/pkg/glue/glue.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ type Session interface {
 	CreateTable(ctx context.Context, dbName model.CIStr, table *model.TableInfo) error
 	CreatePlacementPolicy(ctx context.Context, policy *model.PolicyInfo) error
 	Close()
+	GetGlobalVariables(name string) (string, error)
```
Comment (kennytm) [inline br/pkg/glue/glue.go:41]:
```suggestion
	GetGlobalVariable(name string) (string, error)
```

File: br/pkg/version/version.go [hunk 3]
```diff
@@ -30,6 +30,9 @@ var (
 	compatibleTiFlashMajor4 = semver.New("4.0.0")
 
 	versionHash = regexp.MustCompile("-[0-9]+-g[0-9a-f]{7,}")
+
+	// check whether cross version
+	CrossVesion = false
```
Comment (kennytm) [inline br/pkg/version/version.go:35]: a global mutable variable? i don't think that's a good idea.

**Review by YuJuncen — APPROVED** [review]
LGTM

File: br/pkg/version/version.go (hunk 3 above)
Comment (joccau) [reply br/pkg/version/version.go]: > a global mutable variable? i don't think that's a good idea.

I have remove CrossVersion.

**Pull request merged**